realm/realm-jni/src/io_realm_internal_tableview.cpp

```diff
@@ -75,6 +75,39 @@ JNIEXPORT void JNICALL Java_io_realm_internal_TableView_nativeDistinct(
     } CATCH_STD()
 }
 
+JNIEXPORT void JNICALL Java_io_realm_internal_TableView_nativeDistinctMulti(
+    JNIEnv* env, jobject, jlong nativeViewPtr, jlongArray columnIndexes)
+{
+    if (!VIEW_VALID_AND_IN_SYNC(env, nativeViewPtr))
+        return;
+    try {
+        JniLongArray indexes(env, columnIndexes);
+        jsize indexes_len = indexes.len();
+        std::vector<size_t> columns;
+        for (int i = 0; i < indexes_len; ++i) {
+            if (!COL_INDEX_VALID(env, TV(nativeViewPtr), indexes[i])) {
+                return;
+            }
+            if (!TV(nativeViewPtr)->get_parent().has_search_index(S(indexes[i]))) {
+                ThrowException(env, IllegalArgument, "The field must be indexed before distinct(...) can be used.");
```

Comment:
You are absolutely right. I firstly removed then added back because absence of index check will bring inconsistent behaviors to `distinct*` family. I already have filed an issue.